### PR TITLE
[Headers][X86] Add more descriptions to ia32intrin.h and immintrin.h

### PR DESCRIPTION
**Pull request by pogo59**
ia32intrin.h gets descriptions for all remaining non-privileged intrinsic functions; the macros providing alternate names are not described. immintrin.h ditto, except for the InterlockedExchange functions.

### CODE REVIEW

File: clang/lib/Headers/ia32intrin.h [hunk 1]
```diff
@@ -361,42 +425,146 @@ _wbinvd(void) {
   __builtin_ia32_wbinvd();
 }
 
+/// Rotates an 8-bit value to the left by the specified number of bits.
+///    This operation is undefined if the number of bits exceeds the size of
+///    the value.
+///
+/// \headerfile <x86intrin.h>
+///
+/// This intrinsic corresponds to the \c ROLB instruction.
```
Comment (topperc) [inline clang/lib/Headers/ia32intrin.h:434]: In the Intel and AMD manuals I think the instruction is just called ROL. ROLB is the name in AT&T assembler syntax to distinquish the size.

File: clang/lib/Headers/ia32intrin.h [hunk 2]
```diff
@@ -173,25 +183,59 @@ __popcntq(unsigned long long __A)
 #endif /* __x86_64__ */
 
 #ifdef __x86_64__
+/// Returns the program status and control \c RFLAGS register with the \c VM
+///    and \c RF flags cleared.
+///
+/// \headerfile <x86intrin.h>
+///
+/// This intrinsic corresponds to the \c PUSHFQ + \c POPQ instruction sequence.
```
Comment (phoebewang) [inline clang/lib/Headers/ia32intrin.h:191]: Also `POP` instead `POPQ`. The same below.

Comment (pogo59) [reply clang/lib/Headers/ia32intrin.h]: Done

File: clang/lib/Headers/ia32intrin.h (hunk 1 above)
Comment (pogo59) [reply clang/lib/Headers/ia32intrin.h]: Done

File: clang/lib/Headers/ia32intrin.h [hunk 3]
```diff
@@ -173,25 +183,59 @@ __popcntq(unsigned long long __A)
 #endif /* __x86_64__ */
 
 #ifdef __x86_64__
+/// Returns the program status and control \c RFLAGS register with the \c VM
+///    and \c RF flags cleared.
+///
+/// \headerfile <x86intrin.h>
+///
+/// This intrinsic corresponds to the \c PUSHFQ + \c POP instruction sequence.
```
Comment (topperc) [inline clang/lib/Headers/ia32intrin.h:191]: PUSHFQ->PUSHF

**Review by topperc — APPROVED** [review]
LGTM

File: clang/lib/Headers/ia32intrin.h (hunk 3 above)
Comment (phoebewang) [reply clang/lib/Headers/ia32intrin.h:191]: The mnemonic in SDM is PUSHFQ

**Review by phoebewang — APPROVED** [review]
LGTM.

File: clang/lib/Headers/ia32intrin.h (hunk 3 above)
Comment (topperc) [reply clang/lib/Headers/ia32intrin.h:191]: I wrote that comment before I checked and meant to remove it before I approved. Sorry.

**Pull request merged**